internal/runners/artifacts/artifacts.go

```diff
@@ -362,6 +362,16 @@ func getBuildPlan(
 
 	// Return the artifact map for the given commitID of the given project.
 	case namespaceProvided && commitIdProvided:
+		// Note: the Platform does not raise an error when requesting a commit ID that does not exist in
+		// a given project, so we have verify existence client-side. See DS-1705 (yes, DS, not DX).
+		_, err := model.GetCommitWithinProjectHistory(commitUUID, namespace.Owner, namespace.Project, auth)
+		if err != nil {
+			if err != model.ErrCommitNotInHistory {
+				err = errs.Wrap(err, "Unable to determine if commit exists in project")
+			}
+			return nil, err
+		}
```

Comment:
This isn't just relevant when the namespace is manually provided. I can run `state artifacts --commit` with an invalid commit for my project under CWD or configured globally.

Reply:
Good call!